src/Catalog/CommitCollector.cs

```diff
@@ -71,13 +71,16 @@ public abstract class CommitCollector : CollectorBase
                         Trace.TraceInformation("CommitCatalog.Fetch front.Value saved since timestamp changed from previous: {0}", front);
                     }
 
-                    acceptNextBatch = await OnProcessBatch(
-                        client,
-                        batch.Items.Select(item => item.Value),
-                        context,
-                        batch.CommitTimeStamp,
-                        batch.CommitTimeStamp == lastBatch.CommitTimeStamp,
-                        cancellationToken);
+                    using (_telemetryService.TrackDuration(TelemetryConstants.ProcessBatchSeconds))
```

Comment:
do you think it might be useful to track the number of items in `batch` along with the duration as the metric?(As to average out the sampling per package)

Reply:
It could.  Custom metrics aren't a great fit though as additional numbers need to be stored as a custom dimension (string).